spec.html

```diff
@@ -36004,16 +37158,16 @@ <h1>The Atomics Object</h1>
       <h1>Abstract Operations for Atomics</h1>
 
       <emu-clause id="sec-validatesharedintegertypedarray" aoid="ValidateSharedIntegerTypedArray">
-        <h1>ValidateSharedIntegerTypedArray ( _typedArray_ [ , _onlyInt32_ ] )</h1>
-        <p>The abstract operation ValidateSharedIntegerTypedArray takes one argument _typedArray_ and an optional Boolean _onlyInt32_. It performs the following steps:</p>
+        <h1>ValidateSharedIntegerTypedArray ( _typedArray_ [ , _waitable_ ] )</h1>
+        <p>The abstract operation ValidateSharedIntegerTypedArray takes one argument _typedArray_ and an optional Boolean _waitable_. It performs the following steps:</p>
         <emu-alg>
-          1. If _onlyInt32_ is not present, set _onlyInt32_ to *false*.
+          1. If _waitable_ is not present, set _waitable_ to *false*.
           1. Perform ? RequireInternalSlot(_typedArray_, [[TypedArrayName]]).
           1. Let _typeName_ be _typedArray_.[[TypedArrayName]].
-          1. If _onlyInt32_ is *true*, then
-            1. If _typeName_ is not `"Int32Array"`, throw a *TypeError* exception.
+          1. If _waitable_ is *true*, then
+            1. If _typeName_ is not `"Int32Array"` or `"BigInt64Array"`, throw a *TypeError* exception.
           1. Else,
-            1. If _typeName_ is not `"Int8Array"`, `"Uint8Array"`, `"Int16Array"`, `"Uint16Array"`, `"Int32Array"`, or `"Uint32Array"`, throw a *TypeError* exception.
+            1. If _typeName_ is not `"Int8Array"`, `"Uint8Array"`, `"Int16Array"`, `"Uint16Array"`, `"Int32Array"`, `"Uint32Array"`, `"BigUint64Array"`, or `"BigInt64Array"`, throw a *TypeError* exception.
```

Comment:
is there perhaps a list of these values somewhere so they don't need to be hardcoded in random places?

Reply:
We can use `IsUnclampedIntegerTypedArrayType` and `IsBigIntTypedArrayType` to perform this check without hardcode this list.